pkgs/development/compilers/squeak/default.nix

```diff
@@ -140,7 +164,16 @@ in stdenv.mkDerivation {
   # Workaround build failure on -fno-common toolchains:
   #   ld: vm/vm.a(cogit.o):spur64src/vm/cogitX64SysV.c:2552: multiple definition of
   #       `traceStores'; vm/vm.a(gcc3x-cointerp.o):spur64src/vm/cogit.h:140: first defined here
-  env.NIX_CFLAGS_COMPILE = "-fcommon";
+  env.NIX_CFLAGS_COMPILE = toString (
+    [ "-fcommon" ]
+    ++ (lib.optionals stdenv.cc.isClang [
+      # LLVM 16 turned these into errors (rightly, perhaps.)
+      # Allow this package to continue to build despite this change.
+      "-Wno-error=int-conversion"
+      "-Wno-error=implicit-function-declaration"
+      "-Wno-error=incompatible-function-pointer-types"
+    ])
+  );
```

Comment:
I would prefer this to be a list at `NIX_CFLAGS_COMPILE` rather than a string at `env.NIX_CFLAGS_COMPILE`, unless we have a policy of using `env.«…»` that I don't know about.

Reply:
`env.NIX_CFLAGS_COMPILE` is definitely the majority. I also prefer `NIX_CFLAGS_COMPILE`, and I'll change it to that form as it works just fine.

```
$ rg env.NIX_CFLAGS_COMPILE | wc -l
816

$ rg NIX_CFLAGS_COMPILE | rg -v env | wc -l
236
```

Reply:
No, please don't. Moving environmental variables to `env` is a step towards `__structuredAttrs` support, see #263773.

Reply:
> $ rg NIX_CFLAGS_COMPILE | rg -v env | wc -l
> 236
> ```

Does that count patterns like the following?

```
  env = {
    NIX_CFLAGS_COMPILE = "...";
  };
```

Reply:
It would count those, yes. There's likely quite a few false positives.

Reply:
Believe it because it's an undocumented practice - https://nixos.org/manual/nixpkgs/unstable/.